### PR TITLE
Update go support to 1.20+1.21

### PR DESCRIPTION
**Pull request by fishy**
Also upgrade thrift to 0.19.0.

Also deprecate errorsbp.Batch.

### CODE REVIEW

File: httpbp/errors.go [hunk 1]
```diff
@@ -771,9 +771,9 @@ func ClientErrorFromResponse(resp *http.Response) error {
...
 // keep-alive connections, so you should always defer it after checking error.
 func DrainAndClose(r io.ReadCloser) error {
-	var batch errorsbp.Batch
 	_, err := io.Copy(io.Discard, r)
-	batch.Add(err)
-	batch.Add(r.Close())
-	return batch.Compile()
+	return errors.Join(
+		errorsbp.Prefix("read fully", err),
+		errorsbp.Prefix("close", r.Close()),
```
Comment (pacejackson) [inline httpbp/errors.go:777]: ooc, why do we still need to use errorsbp here?

File: retrybp/retry_test.go [hunk 2]
```diff
@@ -94,23 +94,18 @@ func TestDoBatchError(t *testing.T) {
...
-				"wrong number of errors in %v, expected %v",
-				batchErr.GetErrors(),
-				errList,
-			)
-
-			for _, err := range errList {
-				if !errors.Is(batchErr, err) {
-					t.Errorf("%v is not wrapped by %v", err, batchErr)
-				}
+	if got, want := errorsbp.BatchSize(err), len(errList); got != want {
```
Comment (pacejackson) [inline retrybp/retry_test.go:97]: is there something we can/should replace `errorsbp.BatchSize` with?

**Review by pacejackson — APPROVED** [review]

File: httpbp/errors.go (hunk 1 above)
Comment (fishy) [reply httpbp/errors.go:777]: see https://github.com/reddit/baseplate.go/commit/d939c5374d5c8e453b1cbf5169706b9ac1d6991d

File: retrybp/retry_test.go (hunk 2 above)
Comment (fishy) [reply retrybp/retry_test.go:97]: see https://github.com/reddit/baseplate.go/commit/1edc07c27eff4cfcd00a5ff7bbc69e9d7549d076

basically it's much easier to use than handle a few different cases yourself (check nil, check not a batch error, then try to unwrap to a batch and get the size).

**Review by kylelemons — APPROVED** [review]

**Pull request merged**